integreat_cms/cms/views/pages/page_actions.py

```diff
@@ -827,3 +828,52 @@ def render_mirrored_page_field(request, region_slug, language_slug):
             "page_form": page_form,
         },
     )
+
+
+@require_POST
+@permission_required("cms.change_page")
+# pylint: disable=unused-argument
+def refresh_date(
+    request,
+    page_id,
+    region_slug,
+    language_slug,
+):
+    """
+    Refresh the date for all translations of a corresponding page
+
+    :param request: The current request
+    :type request: ~django.http.HttpResponse
+
+    :param page_id: The id of the page of the current page form
+    :type page_id: int
+
+    :param region_slug: The slug of the current region
+    :type region_slug: str
+
+    :param language_slug: The slug of the current language
+    :type language_slug: str
+
+    :return: A redirection to the :class:`~integreat_cms.cms.views.pages.page_form_view.PageFormView`
+    :rtype: ~django.http.HttpResponseRedirect
+    """
+    region = request.region
+    page = get_object_or_404(region.pages, id=page_id)
+
+    # Consider only the last version of each translation
+    page_translations = page.translations.distinct("page__pk", "language__pk")
+    # Refresh page translations respective their languages depth level in the
+    # language tree, to make sure, that translations won't count as outdated.
+    language_nodes = LanguageTreeNode.objects.filter(region=region).order_by("depth")
```

Comment:
```suggestion

    # Sort page translations according to the position of their languages in the
    # language tree to ensure that the translations are not considered outdated.
    page_translations = sorted(
        page_translations,
        key=lambda page_translation: region.active_languages.index(
            page_translation.language
        ),
    )

```
Instead of querying the language tree nodes and then iterating over it, we could also sort the page translation queryset by the in index of their language in the region's language tree (I noticed that tree nodes are ordered by their `tree_id` and `lft` fields [by default](https://github.com/django-treebeard/django-treebeard/blob/70fd011f5a3bc7fe21653cbdba2c7d6e931aaec2/treebeard/ns_tree.py#L106) so the tree structure is preserved anyway and we can use the prefetched region properties), this way the order is correct, we save a few database queries and we don't have the problem of requesting languages that don't have a corresponding page translation...
(your code would throw the error `'NoneType' object has no attribute 'save'` in this case)